src/components/AttendeeForm.tsx

```diff
@@ -0,0 +1,225 @@
+  championPool: string
+  birthPlace: string
+  avatar: File | null
+}
+type Errors = Partial<Record<keyof Inputs, string>>
+type Touched = Partial<Record<keyof Inputs, boolean>>
+
+const AttendeeForm: React.FC<Props> = ({ tournamentId, onSuscribeOk }) => {
+  const validate = (newInputs: Inputs): Errors => {
+    const newErrors: Errors = {}
+
+    // RiotId
+    const riotIdRegex = /^.{4,16}#[a-zA-Z0-9]{3,5}$/
+    if (!riotIdRegex.test(newInputs.riotId)) {
+      newErrors.riotId = 'Please enter a valid Riot ID'
+    }
+
+    // Comment
+    if (comment.length > 50) {
+      newErrors.comment = 'Please a comment under 50 characters'
+    }
+
+    // BirthPlace
+    if (newInputs.birthPlace === '') {
+      newErrors.birthPlace = 'Please enter a birthplace'
+    }
+
+    // Avatar
+    if (newInputs.avatar === null) {
+      newErrors.avatar = 'Please choose an avatar'
+    }
+
+    return newErrors
+  }
+
+  const [inputs, setInputs] = useState<Inputs>({
+    riotId: '',
+    currentElo: LolElo.values[0],
+    comment: '',
+    role: TeamRole.values[0],
+    championPool: 'One trick poney',
+    birthPlace: '',
+    avatar: null,
+  })
+  const { pb, user } = usePocketBase()
+
+  const [submitError, setSubmitError] = useState<null | string>(null)
+
+  const errors = validate(inputs)
+
+  const [touched, setTouched] = useState<Touched>({})
+
+  const handleChange = (key: keyof Inputs) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputs({ ...inputs, [key]: event.target.value })
+  }
+  const handleSelectChange =
+    (key: keyof Inputs) => (event: React.ChangeEvent<HTMLSelectElement>) => {
+      setInputs({ ...inputs, [key]: event.target.value })
+    }
+
+  const handleBlur = (key: keyof Inputs) => () => {
+    setTouched({ ...touched, [key]: true })
+  }
+
+  const handleFileChange = () => (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files
+    if (files !== null && files.length > 0) {
+      const selected = files[0]
+      setInputs({ ...inputs, avatar: selected })
+    }
+  }
+
+  const showErrorMsg = (key: keyof Inputs): boolean =>
+    errors[key] !== undefined && touched[key] !== undefined
+
+  const handleSubmit = () => (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault()
+
+    let obj = {}
+    for (const cle in inputs) {
+      obj = { ...obj, [cle]: true }
+    }
+
+    setTouched(obj)
+
+    if (Object.keys(errors).length === 0) {
+      if (user !== null) {
+        const data = {
+          ...inputs,
+          isCaptain: false,
+          tournament: tournamentId,
+          user: user.id,
+        }
+
+        pb.collection('attendees')
+          .create(data)
+          .then(() => onSuscribeOk())
+          .catch(() => setSubmitError('An unkwoned error appened'))
+      }
+    }
+  }
+
+  return (
+    <form
+      className="flex w-[26rem] flex-col gap-3 rounded-lg border-2 border-gray-400 bg-white p-4"
+      noValidate={true}
```

Comment:
```suggestion
      onSubmit={handleSubmit}
```
et change le `type` du `button` de soumission à `type="submit"` (et enlève le `onClick` dudit bouton).

Sauf erreur, tu auras besoin d'ajouter un `e.preventDefault()` dans le `handleSubmit`, mais essaye sans.

Reply:
Ok j'ai pas réussi à faire fonctionner avec le bouton passé en type submit 
On verra ça

Reply:
Là partie importante, c'est `onSubmit={handleSubmit}` sur le `<form>`.  
Ça rend le formulaire plus accessible (tu peux le soumettre en appuyant sur Entrer depuis un `<input>`).